### ISSUE TITLE
Different `kernel_size` and `strides` values on `DepthwiseConv2D` (with `padding=same`) will trigger significant inconsistent results across Tensorlfow, CNTK and Theano, even using norml parameters like `kernel_size=2,strides=1`.

### ISSUE THREAD

**Issue by Jup11ter** [keras-team/keras#13842]
**Different `kernel_size` and `strides` values on `DepthwiseConv2D` (with `padding=same`) will trigger significant inconsistent results across Tensorlfow, CNTK and Theano, even using norml parameters like `kernel_size=2,strides=1`.**

## System information

- Have I written custom code (as opposed to using example directory):
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Windows 10 & Linux Ubuntu 18.04
- Tensorflow backend (yes / no): yes
- Tensorflow version：1.14.0-gpu
- Cntk version: 2.7-gpu
- Theano version:1.04-gpu
- Keras version: 2.3.1
- Python version: 3.6.9
- CUDA/cuDNN version: -
- GPU model and memory: -

## Describe the current behavior
**When I use `DepthwiseConv2D` with the parameter `padding=same`, and change the `kernel_size` and `strides`, I notice obvious output differences across 3 backends in some cases**, as shown in the following tables.  (See the codes below to reproduce the issue.)

Note that, **we take the sum of absolute diffenece (SoAD) as the metric** in these tables. (See SoAD implementation in code snippet below.)

1) `kernel_size=1`

**CNTK behaves differently from Tensorflow and Theano in some cases when `kernel_size=1`.**

|strides|Tensorflow-CNTK|Tensorflow-Theano|CNTK-Theano|
| ----: | ---: |---:| ---------: |
|1|  0.    |   0.   |   0. |
|2|  0.    |   0.   |   0. |
|3|  0.    |   0.   |   0. |
|**4**| ***897.31089687*** |   0.   | ***897.31089687*** |
|5| 0.  |  0.      |   0.  |
|6| 0.|  0.        |  0.   |
|**7**|***446.60348872***|  0.  | ***446.60348872*** |
|**8**|***316.42553336***|0.     |***316.42553336***|
|**9**|***273.88436383***| 0.    |***273.88436383***|
|10|  0. |  0.    |  0.  |

2) `kernel_size=2`

**When `strides=1`, three backends show quite different outputs.** Similar issues occur when `strides=4/7/8/9`.  And Theano behaves quite differently from Tensorflow and CNTK in other cases.

|strides|Tensorflow-CNTK|Tensorflow-Theano|CNTK-Theano|
| ----: | ---: |---:| ---------: |
|***1***|***1.66629908e+04***| ***1.89978224e+04*** |***1.94654342e+04***|
|2|3.41665281e-13|4.25164804e+03|4.25164804e+03|
|3|1.90125585e-13| 2.15956529e+03|2.15956529e+03|
|***4***|***1.13031484e+03***| ***1.16128682e+03*** | ***1.22253624e+03*** |
|5|6.84723274e-14|8.51269990e+02| 8.51269990e+02|
|6|4.63077927e-14|6.57431127e+02| 6.57431127e+02|
|***7***|***4.37231640e+02***| ***4.89286548e+02*** |***4.55944453e+02***|
|***8***|***2.53817713e+02***| ***2.63568541e+02*** | ***2.46146475e+02*** |
|***9***|***2.76400556e+02***| ***3.37932890e+02*** | ***3.35156225e+02*** |
|10|2.37899977e-14| 3.07664662e+02| 3.07664662e+02|

3）`kernel_size=3`

**Three backends do not show definitive similarity facing different `strides` values.**

| strides | Tensorflow-CNTK | Tensorflow-Theano | CNTK-Theano    |
| ----: | ---: |---:| ---------: |
|1|2.35979202e-12|3.30831253e-12|2.94574548e-12|
|**2**|***4.64192581e+03***|***4.64192581e+03***| 8.69110122e-13|
|**3**|***2.38899066e+03***| ***2.38899066e+03*** |3.62354989e-13|
|4|1.48203932e-13| ***1.12537857e+03*** | ***1.12537857e+03*** |
|**5**|***9.62017411e+02***| ***9.62017411e+02*** | 1.33954073e-13|
|6|7.45779210e+02| 7.45779210e+02| 9.36620573e-14|
|7|6.25862209e-14| 4.90622760e+02| 4.90622760e+02|
|8|2.70831408e+02| 2.80985657e+02| 2.92329633e+02|
|9|2.56000644e+02| 3.00206323e+02| 3.10693253e+02|
|10|3.37504933e+02| 3.37504933e+02| 4.37499429e-14|

## Key insights

It is explictly mentioned in `Conv2D`'s docs that ***Note that `"same"` is slightly inconsistent across backends with`strides` != 1***,(see `keras\layers\convolutional.py line391` ). **This also applies to the case of `DepthwiseConv2D`,  but we found no similar warnings for `DepthwiseConv2D`**. I think Keras needs to update its comments here. 

Worse still, even when `strides = 1` (e.g., `kernel_size = 2` and `strides = 1`), there shows significant difference across CNTK, Tensorflow and Theano, as shown in the table above.  **Dose it indicate an implementation bug in `DepthwiseConv2D`?  This may lead to potential risk when we use `DepthwiseConv2D` in real applications.** 

## Code to reproduce the issue

```python
import os
import numpy as np
import keras.layers as L
import keras.backend as K
import importlib
from keras.models import load_model
from keras.engine import Model, Input

backends = ['cntk','tensorflow','theano']

# SoAD, Sum of Absolute Difference
def acc_abs_diff(output1, output2):
    assert output1.shape == output2.shape
    abs_diff = np.abs(output1-output2)
    return  np.sum(abs_diff)

def set_keras_backend(backend='tensorflow'):
    if K.backend() != backend:
        os.environ['KERAS_BACKEND'] = backend
        importlib.reload(K.load_backend)
        importlib.reload(K)
        assert K.backend() == backend


kwargs={'kernel_size': 3, 'padding': 'same', 'strides': 1, 'dilation_rate': (1, 1), 'data_format': 'channels_last'}
kwargs['kernel_size']=2
listdiff=[]
for stride_value in range(10):
    kwargs['strides']=stride_value+1
    input_data = (10 * np.random.random((1,32,32,16)))
    input = input_data.astype('float64')
    set_keras_backend('tensorflow')
    layer = L.convolutional.DepthwiseConv2D(**kwargs)# you can also use DepthwiseConv2D
    x = Input(batch_shape=input.shape)
    y = layer(x)
    bk_model = Model(x, y)
    model_path = os.path.join('./', 'model.h5')
    bk_model.save(model_path, bk_model)
    output = {}
    for bk in backends:
        try:
            set_keras_backend(backend=bk)
            model = load_model(model_path)
            output[bk] = model.predict(input)
        except:
            print('error result')
    try:
        diff1 = acc_abs_diff(output['tensorflow'], output['cntk'])
    except:
        diff1 = None
    try:
        diff2=acc_abs_diff(output['theano'], output['cntk'])
    except:
        diff2=None
    try:
        diff3=acc_abs_diff(output['theano'], output['tensorflow'])
    except:
        diff3=None
    listdiff.append([diff1,diff2,diff3])
arraydiff = np.array(listdiff)
print(arraydiff)
print('finish')
```

**Comment by Saduf2019** [comment]
@Jup11ter 
We see that you are using old version of tensorflow which is not actively supported now, please verify on 2.x and open a new issue in case you face any errors, hence moving this to closed status.
